src/urllib3/poolmanager.py

```diff
@@ -166,25 +184,40 @@ class PoolManager(RequestMethods):
     proxy = None
     proxy_config = None
 
-    def __init__(self, num_pools=10, headers=None, **connection_pool_kw):
+    def __init__(
+        self,
+        num_pools: int = 10,
+        headers: Optional[Mapping[str, str]] = None,
+        **connection_pool_kw: Any,
+    ) -> None:
         super().__init__(headers)
         self.connection_pool_kw = connection_pool_kw
-        self.pools = RecentlyUsedContainer(num_pools, dispose_func=lambda p: p.close())
+        func: Callable[[Any], Any] = lambda p: p.close()
+        self.pools: RecentlyUsedContainer[int, Optional[Callable[[_VT], None]]]
+        self.pools = RecentlyUsedContainer(num_pools, dispose_func=func)
 
         # Locally set the pool classes and keys so other PoolManagers can
         # override them.
         self.pool_classes_by_scheme = pool_classes_by_scheme
         self.key_fn_by_scheme = key_fn_by_scheme.copy()
 
-    def __enter__(self):
+    def __enter__(self) -> "PoolManager":
         return self
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
+    def __exit__(
+        self, exc_type: object, exc_val: object, exc_tb: object
+    ) -> "Literal[False]":
```

Comment:
Should we use `bool` here? Does `Literal[False]` have special meaning for mypy + `__exit__`?

Reply:
I think we have to use `Literal[False]`. otherwise we will get:
```
src/urllib3/poolmanager.py:210: error: "bool" is invalid as return type for "__exit__" that always returns False
src/urllib3/poolmanager.py:210: note: Use "typing_extensions.Literal[False]" as the return type or change it to "None"
src/urllib3/poolmanager.py:210: note: If return type of "__exit__" implies that it may return True, the context manager may swallow exceptions

```